api/app.go

```diff
@@ -198,6 +215,8 @@ func (app *App) configureApplication() error {
 	a.Use(NewSentryMiddleware(app).Serve)
 	a.Use(NewNewRelicMiddleware(app, app.Logger).Serve)
 
+	jaeger.InstrumentEcho(a)
```

Comment:
is this needed? isn't there any way to detect and instrument echo instances automagically? how does ddog do this?

Reply:
Not automagically. dd-trace-go has a wrapper of each object he wishes to instrument, which you instantiate instead of the original.

Reply:
It seems that we could achieve better standardization using wrappers, how will it look like when we also instrument postgres for example?

Reply:
For postgres it will be like the redis modifications instead of the echo ones, we'll also make directly it the extensions postgres wrapper. We decided to instrument echo this way because we no longer use the lib in our recent projects. Why would you rather have a wrapper?

Reply:
my only requirement is that it has to be obvious how to use tracing on an extension given how I did with another one, like:
```
redisclient.Trace().Ping()
pgclient.Trace().Select()
```
this is ok!

```
redisclient.Trace().Ping()
pg = jaeger.WrapPG(pgclient)
pg.Select()
```
this is not ok

Reply:
Ok, I think I understood what you said. You think it is not ok to instrument echo implicitly:

```
jaeger.InstrumentEcho(a)
```

but instrument pg and redis explicitly:

```
redisclient.Trace().Ping()
pgclient.Trace().Select()
```

is it correct?